src/components/application_manager/test/message_helper/message_helper_test.cc

```diff
@@ -1058,6 +1058,41 @@ TEST_F(MessageHelperTest,
   }
 }
 
+TEST_F(MessageHelperTest, ProcessSoftButtons_TextAndImageExists_SUCCESS) {
```

Comment:
@indlu, perhaps, it would be better to divide it on three separate tests. And also move all preparations for it, in separate methods